modulemd/v2/modulemd-module.c

```diff
@@ -205,10 +211,20 @@ modulemd_module_get_defaults (ModulemdModule *self)
 void
 modulemd_module_add_stream (ModulemdModule *self, ModulemdModuleStream *stream)
 {
+  ModulemdTranslation *translation = NULL;
+  ModulemdModuleStream *newstream = NULL;
   g_return_if_fail (MODULEMD_IS_MODULE (self));
 
-  g_ptr_array_add (self->streams,
-                   modulemd_module_stream_copy (stream, NULL, NULL));
+  //newstream = modulemd_module_stream_copy (stream, NULL, NULL);
+  newstream = g_object_ref (stream);
+  g_ptr_array_add (self->streams, newstream);
```

Comment:
Why did you move from a copy to a reference here?

Also, it just occurred to me that this function really ought to have
```
g_return_if_fail (g_str_equal(modulemd_module_stream_get_name(stream), self->module_name);
```
since it should really be a coding error if we ever try to add a stream whose name doesn't match the module.

Reply:
The move from copy to ref is a result of the fact that the stream v1 and v2 objects don't yet have their own `copy` implementations, so that internal things (like the summary and description) don't get copied when you copy a stream.
Because of that I moved to a ref during development so I could at least verify that this specific patchset was working.
This patch probably should wait until we've added that, so that the tests can pass with a copy.

Reply:
Aargh, I didn't implement the `copy()`, did I? OK, I'll get that done by the end of Monday...

Reply:
I actually have a fair bit of the `copy()` working for stream v1, so will finish that up and submit by Monday morning.